File: app/src/main/java/com/thetechnocafe/gurleensethi/bottomsheets/BottomSheetFragmentActivity.java
package com.thetechnocafe.gurleensethi.bottomsheets;

import android.support.design.widget.BottomSheetDialogFragment;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class BottomSheetFragmentActivity extends AppCompatActivity {

    public static final String TAG = "bottom_sheet";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_bottom_sheet_fragment);

        Button showBottomSheet = (Button) findViewById(R.id.showSheetButton);

        showBottomSheet.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                BottomSheetFragment fragment = new BottomSheetFragment();
                fragment.show(getSupportFragmentManager(), TAG);
            }
        });
    }
}

File: app/src/main/java/com/thetechnocafe/gurleensethi/bottomsheets/BottomSheetFragmentActivity.kt
package com.thetechnocafe.gurleensethi.bottomsheets

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log
import android.view.View
import android.widget.Button
import kotlinx.android.synthetic.main.activity_bottom_sheet_fragment.*

class BottomSheetFragmentActivity : AppCompatActivity(), OnDataPass {


    var passVariableFromActivityToFrag: Int = 123

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_bottom_sheet_fragment)
        showSheetButton.setOnClickListener {
            val fragment = BottomSheetFragment()
            fragment.show(supportFragmentManager, TAG)
        }
    }

    override fun onDataPass(data: String) {
        Log.d("LOGDATA", "hello $data")
    }
    override fun onDataPass(data: ArrayList<Int>) {
        data.forEach { Log.d("LOGDATA", "hello $it") }
    }

    companion object {

        val TAG = "bottom_sheet"
    }
}
